### PR TITLE
Add EOS Classic network support

### PR DESCRIPTION
**Pull request by eosclassicteam**
EOS Classic is a combination of ETH & EOS protocol network

We would like to request adding EOS Classic Mainnet support on parity-ethereum client

### Coin info

+ Homepage: https://eos-classic.io
+ Explorer: https://explorer.eos-classic.io , https://eosctracker.com ( Now running with parity node )
+ Netstats: https://stats.eos-classic.io
+ Twitter: https://twitter.com/eosclassic.io
+ Chainid and networkid is "0x14"
+ Currently supported by MyEtherWallet, MyCrypto, and Trezor wallet

### Screenshot

Any questions about EOS Classic are welcome! ❤️

### CODE REVIEW

File: ethcore/res/ethereum/classic.json [hunk 1]
```diff
@@ -3077,7 +3077,9 @@
 		"enode://72e445f4e89c0f476d404bc40478b0df83a5b500d2d2e850e08eb1af0cd464ab86db6160d0fde64bd77d5f0d33507ae19035671b3c74fec126d6e28787669740@104.198.71.200:30303",
 		"enode://39abab9d2a41f53298c0c9dc6bbca57b0840c3ba9dccf42aa27316addc1b7e56ade32a0a9f7f52d6c5db4fe74d8824bcedfeaecf1a4e533cacb71cf8100a9442@144.76.238.49:30303",
 		"enode://f50e675a34f471af2438b921914b5f06499c7438f3146f6b8936f1faeb50b8a91d0d0c24fb05a66f05865cd58c24da3e664d0def806172ddd0d4c5bdbf37747e@144.76.238.49:30306",
-		"enode://83b33409349ffa25e150555f7b4f8deebc68f3d34d782129dc3c8ba07b880c209310a4191e1725f2f6bef59bce9452d821111eaa786deab08a7e6551fca41f4f@159.89.223.6:30303"
+		"enode://83b33409349ffa25e150555f7b4f8deebc68f3d34d782129dc3c8ba07b880c209310a4191e1725f2f6bef59bce9452d821111eaa786deab08a7e6551fca41f4f@159.89.223.6:30303",
+		"enode://5cd218959f8263bc3721d7789070806b0adff1a0ed3f95ec886fb469f9362c7507e3b32b256550b9a7964a23a938e8d42d45a0c34b332bfebc54b29081e83b93@35.187.57.94:30303",
+		"enode://6dd3ac8147fa82e46837ec8c3223d69ac24bcdbab04b036a3705c14f3a02e968f7f1adfcdb002aacec2db46e625c04bf8b5a1f85bb2d40a479b3cc9d45a444af@104.237.131.102:30303"
```
Comment (debris) [inline ethcore/res/ethereum/classic.json]: why is this pr changing bootnode addresses of unrelated chains?

Comment (eosclassicteam) [reply ethcore/res/ethereum/classic.json]: @debris Cause this is what we call open source, I can revert the commit but as you know there is no fixed maintainer of these chains, all bootnodes are being updated from parity team or volunteers like me.

I've updated the list based on these list from their chain client, you can refer them here

ETC: https://raw.githubusercontent.com/ethereumproject/go-ethereum/master/core/config/mainnet_bootnodes.json

Ropsten: https://raw.githubusercontent.com/ethereumproject/go-ethereum/master/core/config/morden_bootnodes.json

ELLA: https://raw.githubusercontent.com/ellaism/parity-config/master/ellaism.json

EXP: https://raw.githubusercontent.com/expanse-org/go-expanse/master/params/bootnodes.go

MUSIC: https://raw.githubusercontent.com/Musicoin/go-musicoin/master/params/bootnodes.go

Comment (eosclassicteam) [reply ethcore/res/ethereum/classic.json]: @debris So should I revert the commit and seperate the PR??

Comment (debris) [reply ethcore/res/ethereum/classic.json]: It would be great 🙌

Comment (debris) [reply ethcore/res/ethereum/classic.json]: This will allow us to easily backport bootnodes update to stable and beta

Comment (eosclassicteam) [reply ethcore/res/ethereum/classic.json]: @debris Seperated the PR for bootnode update 😄

**Comment by eosclassicteam** [comment]
@debris Also squashed commit for better future review

File: ethcore/src/ethereum/ethash.rs [hunk 2]
```diff
@@ -154,6 +176,17 @@ impl From<ethjson::spec::EthashParams> for EthashParams {
...
+			eosc_treasury_address: p.eosc_treasury_address.map_or_else(Address::new, Into::into),
+			eosc_treasury_reward: p.eosc_treasury_reward.map_or_else(Default::default, Into::into),
+			eosc_stake_address: p.eosc_stake_address.map_or_else(Address::new, Into::into),
+			eosc_stake_reward: p.eosc_stake_reward.map_or_else(Default::default, Into::into),
+			neweosc_transition: p.neweosc_transition.map_or(u64::max_value(), Into::into),
+			neweosc_pow_reward: p.neweosc_pow_reward.map_or_else(Default::default, Into::into),
+			neweosc_fund_address: p.neweosc_fund_address.map_or_else(Address::new, Into::into),
+			neweosc_fund_reward: p.neweosc_fund_reward.map_or_else(Default::default, Into::into),
+			neweosc_pos_address: p.neweosc_pos_address.map_or_else(Address::new, Into::into),
+			neweosc_pos_reward: p.neweosc_pos_reward.map_or_else(Default::default, Into::into),
```
Comment (5chdn) [inline ethcore/src/ethereum/ethash.rs]: What is `neweosc`?

File: parity/params.rs [hunk 3]
```diff
@@ -43,6 +43,7 @@ pub enum SpecType {
 	Ellaism,
 	Easthub,
 	Social,
+	EOSC,
```
Comment (5chdn) [inline parity/params.rs]: "EOS Classic"

File: ethcore/src/ethereum/ethash.rs (hunk 2 above)
Comment (eosclassicteam) [reply ethcore/src/ethereum/ethash.rs]: @5chdn `neweosc` is our code name of Anti-inflation Hardfork on block num 150,000 please refer here https://github.com/eosclassic/go-eosclassic/issues/5

File: parity/params.rs (hunk 3 above)
Comment (eosclassicteam) [reply parity/params.rs]: Ethereum Classic is using "Ethereum Classic" on their chain spec json, and "Classic" on parity/params.rs file.

It is same with "Ethereum Social" also so I would like to ask using "EOS Classic" on our json file and "EOSC" here 😄

**Comment by eosclassicteam** [comment]
@5chdn @debris @sorpaas When merge?? 😂

**Comment by sorpaas** [comment]
@eosclassicteam My only comment is that you would need to fix ECIP1017 first. Otherwise this would be a huge maintenance burden for all of us.

**Comment by eosclassicteam** [comment]
@sorpaas Well as I've already said right now we don't have enough man power to fix ECIP1017 for our implementation so if anyone is interested on our PR I will reward him as a small bounty in EOSC 😂

**Comment by 5chdn** [comment]
Please, reopen this when you have time to fix the consensus gap between the two clients,

**Pull request closed without merging**